controllers/kubevirtmachine_controller.go

```diff
@@ -587,6 +584,11 @@ func (r *KubevirtMachineReconciler) reconcileKubevirtBootstrapSecret(ctx *contex
 		return errors.New("error retrieving bootstrap data: secret value key is missing")
 	}
 
+	if bootstrapDataSecret != nil && bytes.Compare(bootstrapDataSecret.Data["value"], value) == 0 {
+		ctx.BootstrapDataSecret = bootstrapDataSecret
+		return nil
+	}
```

Comment:
isn't this similar to what the `CreateOrUpdate` function is supposed to be doing?